### ISSUE TITLE
Add support for Converting uint16 TIFF Images to PyTorch Tensors.

### ISSUE THREAD

**Issue by RobuRishabh** [pytorch/pytorch#130310]
### 🚀 The feature, motivation and pitch

I'm working on a dataset consisting of .raw images that I've processed and converted into 16-bit TIFF images (uint16). However, I encountered a challenge when trying to convert these 16-bit TIFF images into tensors for further processing in PyTorch.

The problem arises because the standard PyTorch tensor creation functions (like torch.tensor()) do not natively support the uint16 data type. PyTorch typically supports tensor creation for common data types such as float32, int64, and uint8, but uint16 is not among the directly supported types.

For my research, it is crucial to maintain the uint16 data type of the input images because altering the data type (e.g., converting to float32 or uint8) could lead to loss of precision or information, which could adversely affect the outcomes of my experiments.

**Comment by drisspg** [comment]
I cant reproduce this

```Python
>>> import torch
>>> a = torch.tensor([1,2], dtype=torch.uint16)
>>> a
tensor([1, 2], dtype=torch.uint16)
```
